ceph/ceph_admin/helper.py

```diff
@@ -543,6 +543,23 @@ def generate_ingress_spec(self, spec):
         if node_names:
             spec["placement"]["hosts"] = self.get_hostnames(node_names)
 
+        if spec["spec"].get("ssl_cert") == "create-cert":
+            subject = {
+                "common_name": spec["placement"]["hosts"][0],
+                "ip_address": self.cluster.get_node_by_hostname(
+                    spec["placement"]["hosts"][0]
+                ).ip_address,
+            }
+            key, cert, ca = generate_self_signed_certificate(subject=subject)
+            pem = key + cert + ca
+            cert_value = "|\n" + pem
+            spec["spec"]["ssl_cert"] = "\n    ".join(cert_value.split("\n"))
+            LOG.debug(pem)
+            vip = self.cluster.get_node_by_hostname(
+                spec["placement"]["hosts"][0]
+            ).ip_address
+            spec["spec"]["virtual_ip"] = vip + "/22"
```

Comment:
can we try getting CIDR dynamically.

Reply:
let's avoid `/22` CIDR, so this could help us adopt any system.

Reply:
Done , pls review the latest